### DISCUSSION TITLE
Has the setup process been researched to skip the cloud?

### DISCUSSION THREAD

**Discussion by yoshimo** [make-all/tuya-local#1237] [General]
I was wondering how the setup phase works when you use original devices and the smart life app.
Has anyone looked into it to see if we can pair our devices directly with HA and skip the cloud process without reflashing and tearing apart our hardware?

**Comment by make-all** [comment]
You can't. To pair without the cloud, you need to reflash the devices with different firmware, like ESPHome or Tasmota.

**Reply by yoshimo** [reply]
That was fast and void of details.
Which part of the process prevents us from doing this?
We can listen to transmissions on the air, bluetooth, wifi, every other frequency. Is there some public private key auth that we can't replicate?
